activerecord/lib/active_record/relation/predicate_builder/relation_handler.rb

```diff
@@ -12,6 +12,10 @@ def call(attribute, value)
           value = value.select(value.arel_attribute(value.klass.primary_key))
         end
 
+        value.optimizer_hints_values = value.optimizer_hints_values.select do |hint|
```

Comment:
If we have a hint `"MAX_EXECUTION_TIME(1000) NO_RANGE_OPTIMIZATION(posts index_posts_on_author_id)"`, do we want to remove `"NO_RANGE_OPTIMIZATION(posts index_posts_on_author_id)"` as well in the subquery?

Reply:
You're right, I haven't thought about that case.
I assumed that two different hints would be two items of the values array - e.g. `optimizer_hints("MAX_EXECUTION_TIME(1000)").optimizer_hints("NO_RANGE_OPTIMIZATION(posts)")`